test/integration/resourcemanager/managedresource/reconciler_test.go

```diff
@@ -254,6 +254,226 @@ var _ = Describe("ManagedResource controller tests", func() {
 			})
 		})
 	})
+
+	Context("#Reconciliation Modes/Annotations", func() {
+		BeforeEach(func() {
+			configMap = &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: corev1.SchemeGroupVersion.String(),
+					Kind:       "ConfigMap",
+				}, ObjectMeta: metav1.ObjectMeta{
+					Name:      configMapName,
+					Namespace: namespaceName,
+				}}
+
+			secretForManagedResource = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      secretName,
+					Namespace: namespaceName,
+				},
+			}
+
+			managedResource = &resourcesv1alpha1.ManagedResource{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-managedresource",
+					Namespace: namespaceName,
+				},
+				Spec: resourcesv1alpha1.ManagedResourceSpec{
+					Class:      pointer.String(filter.ResourceClass()),
+					SecretRefs: []corev1.LocalObjectReference{{Name: secretForManagedResource.Name}},
+				},
+			}
+
+			DeferCleanup(func() {
+				Expect(testClient.Delete(ctx, managedResource)).To(Or(Succeed(), BeNotFoundError()))
+				Eventually(func(g Gomega) {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)).To(BeNotFoundError())
+				}, time.Minute, 5*time.Second).Should(Succeed())
+
+				Expect(testClient.Delete(ctx, secretForManagedResource)).To(Or(Succeed(), BeNotFoundError()))
+			})
+		})
```

Comment:
Can't we reuse these code here?
https://github.com/gardener/gardener/blob/190bceffa4415cbe5eda12cd8fe878565b37a53e/test/integration/resourcemanager/managedresource/reconciler_test.go#L53-L85
We would have to adapt some code, of course.

Reply:
I have kept them separately so the contexts are clean, and the code is more readable.

Reply:
hmm.. I still think we can prevent repeating these lines.